src/shared/api/validation.ts

```diff
@@ -258,39 +259,75 @@ export function validate(
-        });
-    }
-
-    return chXHR
-        .then(({ body, elapsedTime }: any) => {
-            let legacyUrl = url.replace(/column-store\//, '');
-            if (treatmentLegacyUrl[label]) {
-                legacyUrl = treatmentLegacyUrl[label](legacyUrl);
+        if (assertResponse && assertResponse[0] && assertResponse[0].attributeId) {
+            let data: any = {};
+            data['attributes'] = [{ attributeId: assertResponse[0].attributeId }];
+            if (params.studyViewFilter) {
+                const {
+                    attributes,
+                    ...filteredStudyViewFilter
+                } = params.studyViewFilter;
+                data['studyViewFilter'] = filteredStudyViewFilter;
+            } else {
+                data['studyViewFilter'] = params;
             }
-            const legacyXHR = $.ajax({
+            chXHR = $.ajax({
                 method: 'post',
-                url: legacyUrl,
+                url: '/api/column-store/clinical-data-counts/fetch?',
+                data: JSON.stringify(data),
+                contentType: 'application/json',
+            }).then((body, state, xhr) => {
+                return { body, elapsedTime: xhr.getResponseHeader('elapsed-time') };
+            });
+        } else {
+            chXHR = $.ajax({
+                method: 'post',
+                url: url,
                 data: JSON.stringify(params),
                 contentType: 'application/json',
+            }).then((body, state, xhr) => {
+                return { body, elapsedTime: xhr.getResponseHeader('elapsed-time') };
             });
-            return legacyXHR.then(legacyResult => {
-                const result: any = compareCounts(body, legacyResult, label);
+        }
+    }
+
+    return chXHR
+        .then(({ body, elapsedTime }: any) => {
+            if (!assertResponse) {
+                let legacyUrl = url.replace(/column-store\//, '');
+                if (treatmentLegacyUrl[label]) {
+                    legacyUrl = treatmentLegacyUrl[label](legacyUrl);
+                }
+                const legacyXHR = $.ajax({
+                    method: 'post',
+                    url: legacyUrl,
+                    data: JSON.stringify(params),
+                    contentType: 'application/json',
+                });
+                return legacyXHR.then(legacyResult => {
+                    const result: any = compareCounts(body, legacyResult, label);
+                    result.url = url;
+                    result.hash = hash;
+                    result.data = params;
+                    result.chDuration = parseFloat(elapsedTime);
+                    result.legacyDuration = parseFloat(
+                        legacyXHR.getResponseHeader('elapsed-time') || ''
+                    );
+                    return result;
+                });
+            }
+            else {
+                const result: any = compareCounts(
```

Comment:
the code to build the result is redundant.  there should be a way of avoiding it.  e.g., something like around 301, if there's an assertResponse, then legacyXHR is just Promise.resolve(assertReseponse).then(...)

Reply:
you will have to figure out a way of normalizing assertResponse and the XHR.  because obviously assertResponse doesn't have getResponseHeader.  That should be easy though.